server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -1589,6 +1509,120 @@ static String resolveExpression(String expression, LongSupplier getTime) {
         }
     }
 
+    public static final class ExplicitResourceNameFilter {
+
+        private ExplicitResourceNameFilter() {
+            // Utility class
+        }
+
+        public static List<String> filter(Context context, List<String> expressions) {
```

Comment:
nit: can you add some java doc for what is contained in the list after this method ? (it is not obvious that exclusions (with the leading -) nor wildcard expressions will be returned in the list ).  maybe this method should be named filterUnavailable ?

Reply:
Pushed [4b81818](https://github.com/elastic/elasticsearch/pull/92518/commits/4b818184a507cc30d06f302a27d363d4058ba35e)